base/src/Data/Macaw/AbsDomain/JumpBounds.hs

```diff
@@ -386,8 +386,18 @@ exprRangePred info e = do
             -- Compare the range constraint with the output number of bits.
             -- If the bound on r covers at most the truncated number
             -- of bits, then we just pass it through.
-          , Just LeqProof <- testLeq (rangeWidth r) w ->
-            SomeRangePred r
+          -> case testLeq (rangeWidth r) w of
+              Just LeqProof -> SomeRangePred r
+              
+              -- Otherwise, our bound remains valid for the truncated number
+              -- of bits and we can tighten the bounds to at most @2^w-1@ (inclusive)
+              Nothing ->
+                let 
+                  truncUnsigned = fromInteger . unsignedClamp w . toInteger
```

Comment:
Can you add an explicit `Natural -> Natural` type signature to `truncUnsigned`? The `fromInteger`/`toInteger` functions work over many types, so without a type signature, you'd have to look elsewhere to realize what type it is actually being used on.

Reply:
Done! I rewrote the commit.